firefox-ios/Ecosia/Account/Auth/Auth0ProviderProtocol.swift

```diff
@@ -44,5 +53,50 @@ public protocol Auth0ProviderProtocol {
     ///
     /// - Returns: A `Credentials` object upon successful renewal.
     /// - Throws: An error if the credential renewal fails.
+    @discardableResult
     func renewCredentials() async throws -> Credentials
 }
+
+extension Auth0ProviderProtocol {
+
+    /// - Returns: An HTTPS `WebAuth`
+    func makeHttpsWebAuth() -> WebAuth {
+        Auth0
+            .webAuth(bundle: .ecosia)
+            .useHTTPS()
+    }
+}
+
+/// Default Protocol Implementation
+extension Auth0ProviderProtocol {
```

Comment:
Nice idea with the default implementation 👏 

I wonder why `webAuth` is not also here? Seems like it could also be a default value and then be left out of `WebAuth0Provider` too.

Reply:
Can add it ofc 💪. It's purely a design decision. Wanted to make it explicit per each provider as it's the most prominent property of an entire Auth0's Web-based authentication.

Reply:
Sure, makes sense, can be left out 🙂